java/code/src/com/redhat/rhn/common/hibernate/HibernateFactory.java

```diff
@@ -379,6 +381,31 @@ public static void commitTransaction() throws HibernateException {
         connectionManager.commitTransaction();
     }
 
+    /**
+     * @param committed - if it was possible to commit the transaction.
+     * Roll back transaction in case it is not committed and close the Hibernate session.
+     */
+    public static void finishTransaction(boolean committed) {
```

Comment:
I'm not sure about this method here. It's not also clear what the method does. The name is finish transaction, receives a parameter named committed, but it doesn't commit anything.
Maybe it could change this method and not receive any parameter, and we perform the commit and rollback logic here (or another helper class) and remove it from the existing caller.
The exception handling can be done outside, since each caller have is logic.

Reply:
I understood that "commited" is the info if the commit was successful or not. I think the behavior differ if it was successful or not.

Reply:
sure, but why not do the commit here, if the logic is the same in both places, in this way we treat the errors in a central place as the commit.
Also, the method name looks like it will deal with all, but it doesn't, and if you don't read what the parameter description says, you can be fooled. 
One of the biggest problem in computer science: naming stuff :)